src/Frequency.zig

```diff
@@ -10,8 +10,7 @@ target: u32 = 0,
 /// The estimated delay that is needed to achieve the target frequency. Updated during tick()
 delay_ns: u64 = 0,
 
-/// The actual measured frequency. This is updated every 1/10th second (the number of actual ticks
-/// measured every 1/10th second, multiplied by 10.)
+/// The actual measured frequency. This is updated every second.
 rate: u32 = 0,
```

Comment:
I don't have a way to verify your refactored code is correct (as we have no tests for this logic yet), and some of the changes here like this doc comment lead me to believe it might not be - or that you may be changing behavior.

I think we need more tests around this logic before we can refactor it, but if you want to revise this PR to just be the most minimal set of changes needed for:

> The reset logic was broken, it would set `delta_time = 0` every second, meaning every second would be missing the timings for one frame.

then I can consider that change in isolation.

Hope that makes sense!

Reply:
I haven't changed any behaviour; that comment was incorrect and I changed it to match the actual behaviour of the code (which is the same both before and after my changes).

I did also do one small refactor (changing an `if` to `@min`) which I can revert if you prefer; all other changes are required in order to fix the bug.